src/main/java/io/cosmosoftware/kite/steps/OpenURLStep.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) CoSMo Software Consulting Pte. Ltd. - All Rights Reserved
+ */
+
+package io.cosmosoftware.kite.steps;
+
+import io.cosmosoftware.kite.exception.KiteTestException;
+import io.cosmosoftware.kite.interfaces.Runner;
+import io.cosmosoftware.kite.report.Status;
+import java.util.ArrayList;
+import java.util.List;
+
+public class OpenURLStep extends TestStep {
+
+  private final String url;
+  private final List<String> errorMessages = new ArrayList<>();
+
+
+
+  public OpenURLStep(Runner runner, String url) {
+    super(runner);
+    this.url = url;
+
+    errorMessages.add("Unable to connect");
+    errorMessages.add("Secure Connection Failed");
+    errorMessages.add("The connection has timed out");
+    errorMessages.add("We’re having trouble finding that site");
+
+    errorMessages.add("This site can’t be reached");
+    errorMessages.add("webpage is not available");
+    errorMessages.add("404. That’s an error.");
+  }
+
+  @Override
+  public String stepDescription() {
+    return "Opening: " + this.url;
+  }
+
+  @Override
+  protected void step() throws KiteTestException {
+    this.webDriver.get(this.url);
+    String html = webDriver.getPageSource();
+    for (String errorMessage : this.errorMessages) {
+      if (html.contains(errorMessage)) {
```

Comment:
it will throw exception if i want to load this page:
https://support.mozilla.org/en-US/kb/websites-dont-load-troubleshoot-and-fix-errors#firefox:win10:fx69

or this page:
https://community.cloudflare.com/t/help-we-re-having-trouble-finding-that-site-we-can-t-connect-to-the-server/53230

what are you trying to do exactly?

Reply:
Webdriver has no access to http response code, so I thought I could rely on  webpage content. At least for the tests that we are doing, the error messages only appear if there's an error

Reply:
make this optional, configurable.

Reply:
This is just a predefined OpenURLStep that checks with a list of basic error messages, if you want to check your way then you can always create you own step and use the criteria you want right?